src/overlays/actors/ovl_En_Dekunuts/z_en_dekunuts.c

```diff
@@ -60,59 +60,474 @@ static ColliderCylinderInit D_809EAB50 = {
+        spawnPos.y = this->actor.world.pos.y + 12.0f;
+        spawnPos.z = this->actor.world.pos.z + (Math_CosS(this->actor.shape.rot.y) * 23.0f);
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_NUTSBALL, spawnPos.x, spawnPos.y, spawnPos.z,
+                        this->actor.shape.rot.x, this->actor.shape.rot.y, this->actor.shape.rot.z, 0) != NULL) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_THROW);
+        }
+    } else if ((this->animFlagAndTimer > 1) && Animation_OnFrame(&this->skelAnime, 12.0f)) {
+        Animation_MorphToPlayOnce(&this->skelAnime, &gDekuNutsSpitAnim, -3.0f);
+        if (this->animFlagAndTimer != 0) {
+            this->animFlagAndTimer--;
+        }
+    }
+}
+
+void EnDekunuts_Burrow(EnDekunuts* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnDekunuts_SetupWait(this);
+    } else {
+        this->collider.dim.height = ((3.0f - CLAMP(this->skelAnime.curFrame, 1.0f, 3.0f)) * 12.0f) + 5.0f;
+    }
+    if (Animation_OnFrame(&this->skelAnime, 4.0f)) {
+        this->collider.base.acFlags &= ~AC_ON;
+    }
+    Math_ApproachF(&this->actor.world.pos.x, this->actor.home.pos.x, 0.5f, 3.0f);
+    Math_ApproachF(&this->actor.world.pos.z, this->actor.home.pos.z, 0.5f, 3.0f);
+}
+
+void EnDekunuts_BeginRun(EnDekunuts* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->runDirection = this->actor.yawTowardsPlayer + 0x8000;
+        this->runAwayCount = 3;
+        EnDekunuts_SetupRun(this);
+    }
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 0xE38);
+}
+
+void EnDekunuts_Run(EnDekunuts* this, GlobalContext* globalCtx) {
+    s16 diffRotInit;
+    s16 diffRot;
+    f32 phi_f0;
+
+    SkelAnime_Update(&this->skelAnime);
+    if (Animation_OnFrame(&this->skelAnime, 0.0f) && (this->animFlagAndTimer != 0)) {
+        this->animFlagAndTimer--;
+    }
+    if (this->playWalkSound) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_WALK);
+        this->playWalkSound = false;
+    } else {
+        this->playWalkSound = true;
+    }
+
+    Math_StepToF(&this->actor.speedXZ, 7.5f, 1.0f);
+    if (Math_SmoothStepToS(&this->actor.world.rot.y, this->runDirection, 1, 0xE38, 0xB6) == 0) {
+        if (this->actor.bgCheckFlags & 0x20) {
+            this->runDirection = Actor_WorldYawTowardPoint(&this->actor, &this->actor.home.pos);
+        } else if (this->actor.bgCheckFlags & 8) {
+            this->runDirection = this->actor.wallYaw;
+        } else if (this->runAwayCount == 0) {
+            diffRotInit = Actor_WorldYawTowardPoint(&this->actor, &this->actor.home.pos);
+            diffRot = diffRotInit - this->actor.yawTowardsPlayer;
+            if (ABS(diffRot) >= 0x2001) {
+                this->runDirection = diffRotInit;
+            } else {
+                phi_f0 = (0.0f <= (f32)diffRot) ? 1.0f : -1.0f;
```

Comment:
```suggestion
                phi_f0 = (diffRot >= 0.0f) ? 1.0f : -1.0f;
```
Literal goes on the right. Also I think just having the `0.0f` forces the float conversion.